### PR TITLE
Add DebugMiddleware to debug actions and state changes.

### LINKED ISSUE #27: Add debug code
Implement enable/disable redukt debug code.
Create logger middleware.

### PR DESCRIPTION
**Pull request by raulccabreu**
- Create DebugMiddleware class.
- Add DebugMiddleware instance to middlewares and listeners
at Redukt start if debug mode on.
- Add debug flag at Redukt constructor with default false.
- Unify time spent and new debug flags.
- Prettify time spent log message.
- Update gradle and kotlin dependencies version.
- Change redukt version to the latest release at README import
instructions.

Close #27

### CODE REVIEW

File: app/src/main/java/com/github/raulccabreu/redukt/middlewares/DebugMiddleware.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package com.github.raulccabreu.redukt.middlewares
+
+import com.github.raulccabreu.redukt.actions.Action
+import com.github.raulccabreu.redukt.states.StateListener
+
+/**
+ * Created by raulcca on 12/15/17.
```
Comment (flaviokreis) [inline app/src/main/java/com/github/raulccabreu/redukt/middlewares/DebugMiddleware.kt]: You can remove unnecessary comments.

File: app/src/main/java/com/github/raulccabreu/redukt/middlewares/DebugMiddleware.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+class DebugMiddleware<T> : Middleware<T>, StateListener<T> {
+    override fun before(state: T, action: Action<*>) {
+        println("------------------------------------------------------------")
```
Comment (flaviokreis) [inline app/src/main/java/com/github/raulccabreu/redukt/middlewares/DebugMiddleware.kt]: You can remove this print.

**Review by flaviokreis — CHANGES_REQUESTED** [review]
This is good to test the app.

Is it good use the println, or use the Log.d ?

**Comment by raulccabreu** [comment]
@flaviokreis review changes maded!

**Review by niltonvasques — APPROVED** [review]
Really nice!

**Review by flaviokreis — APPROVED** [review]

**Pull request merged**